### DISCUSSION TITLE
With React 19 rc, is next 15 ready for prod?

### DISCUSSION THREAD

**Discussion by Godrules500** [vercel/next.js#72769] [Help]
### Summary

I'm trying to determine if my production application should be upgraded to next 15 since react 19 is still in RC status?

**Comment by icyJoseph** [comment]
Over at the Next.js reddit, the general sentiment is, if you don't have a rush, then stay put, but not upgrade just yet. 

It is not just React, nor Next.js that you need to think about, but the other set of libraries that you use. Those have also got to upgrade to React 19 compatibility. For example, https://ui.shadcn.com/docs/react-19, just recently managed to get React 19

**Reply by Godrules500** [reply]
Sounds good. There's just a new shiny red button that says upgrade!  What should I look for to know that it is time?

**Reply by icyJoseph** [reply]
Mostly the official React 19 release, and that libraries you use are also compatible with React 19. 

I'd say though, there's a code mod from the Next team, to upgrade your stuff. It might be worth seeing what happens when you run it on your codebase. There are a few breaking changes to be aware of. 

- https://nextjs.org/blog/next-15#smooth-upgrades-with-nextcodemod-cli
